libraries/browser-tracker-core/src/tracker/types.ts

```diff
@@ -581,3 +586,48 @@ export interface BrowserTracker {
    */
   addPlugin: (configuration: BrowserPluginConfiguration) => void;
 }
+
+/**
+ * Schema for client client session context entity
+ */
+export interface ClientSession extends Record<string, unknown> {
```

Comment:
This `extends Record<string, unknown>` thing is a really weird requirement. We should make `PayloadBuilder.addContextEntity()` generic at some point.

```diff
-  addContextEntity: (entity: SelfDescribingJson) => void;
+  addContextEntity: <T extends Record<keyof T, unknown>>(entity: SelfDescribingJson<T>) => void;
```

Reply:
Yes, I have been eyeing that for many months now.